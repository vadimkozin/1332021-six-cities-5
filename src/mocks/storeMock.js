import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
