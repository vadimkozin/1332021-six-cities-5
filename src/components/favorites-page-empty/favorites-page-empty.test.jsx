import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import {render} from 'enzyme';
import storeMock, {offers} from '../../mocks/storeMock';
import FavoritesPageEmpty from './favorites-page-empty';

describe(`FavoritesPageEmpty`, () => {
  it(`should render FavoritesPageEmpty component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <MemoryRouter>
            <FavoritesPageEmpty offers={offers}/>
          </MemoryRouter>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
