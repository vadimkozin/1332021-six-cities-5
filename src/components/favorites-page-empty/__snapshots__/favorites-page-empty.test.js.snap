// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPageEmpty should render FavoritesPageEmpty component 1`] = `
<div
  class="page page--favorites-empty"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/login"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  class="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      class="page__favorites-container container"
    >
      <section
        class="favorites favorites--empty"
      >
        <h1
          class="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          class="favorites__status-wrapper"
        >
          <b
            class="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            class="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    class="footer"
  >
    <a
      class="footer__logo-link"
      href="/"
    >
      <img
        alt="6 cities logo"
        class="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
